Doc/reference/executionmodel.rst

```diff
@@ -139,8 +139,9 @@ namespace.  Names are resolved in the top-level namespace by searching the
 global namespace, i.e. the namespace of the module containing the code block,
 and the builtins namespace, the namespace of the module :mod:`builtins`.  The
 global namespace is searched first.  If the names are not found there, the
-builtins namespace is searched.  The :keyword:`!global` statement must precede
-all uses of the listed names.
+builtins namespace is searched.  A new variable is created in the global 
```

Comment:
This whole paragraph is confusing to read, it might need a clearer separation between what happens when a variable is read versus written. I'll need to think about it more.

Reply:
@JelleZijlstra Do you have suggestion on how else to rewrite it?

Reply:
@JelleZijlstra I have moved a line and made minor changes to the paragraph. Is it clearer now?

The global statement must precede all uses of the listed names. If the global statement occurs within a block, all uses of the names specified in the statement refer to the bindings of those names in the top-level namespace. Names are resolved in the top-level namespace by searching the global namespace, the namespace of the module containing the code block, and the builtins namespace, the namespace of the module builtins. The global namespace is searched first. If the names are not found there, the builtins namespace is searched. If the name is also not found in the builtins namespace, a new variable is created in the global namespace.

Reply:
This is better but I realized it now switches from plural to singular in the last sentence.

Reply:
Fixed!